cvat/apps/dataset_manager/bindings.py

```diff
@@ -966,30 +967,48 @@ def _make_image(i, **kwargs):
                         return _make_image
                 else:
                     # for images use encoded data to avoid recoding
-                    def image_maker_factory(frame_provider):
+                    def image_maker_factory(task):
+                        frame_provider = FrameProvider(task.data)
                         def _make_image(i, **kwargs):
                             loader = lambda _: frame_provider.get_frame(i,
                                 quality=frame_provider.Quality.ORIGINAL,
                                 out_type=frame_provider.Type.BUFFER)[0].getvalue()
                             return ByteImage(data=loader, **kwargs)
                         return _make_image
-                image_maker_per_task[task.id] = image_maker_factory(frame_provider)
+                image_maker_per_task[task.id] = image_maker_factory(task)
 
         for frame_data in project_data.group_by_frame(include_empty=True):
             image_args = {
                 'path': frame_data.name + ext_per_task[frame_data.task_id],
                 'size': (frame_data.height, frame_data.width),
             }
-            if include_images:
+            if self._dimension == DimensionType.DIM_3D:
+                dm_image = image_maker_per_task[frame_data.task_id](frame_data.id, **image_args)
+            elif include_images:
                 dm_image = image_maker_per_task[frame_data.task_id](frame_data.idx, **image_args)
```

Comment:
Why one line uses `frame_data.id` and another one `frame_data.idx`?

Reply:
Because `make_image` functions are different depends on task type and dimension, and implementation of `make_image` function comes from some of 3D feature milestones.